WordPress/Classes/ViewRelated/Me/AppIconViewController.swift

```diff
@@ -0,0 +1,151 @@
+        tableView.rowHeight = Constants.rowHeight
+    }
+
+    // MARK: - UITableview Data Source
+
+    open override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    open override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return icons.count
+    }
+
+    open override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let icon = icons[indexPath.row]
+
+        let cell = tableView.dequeueReusableCell(withIdentifier: Constants.cellIdentifier, for: indexPath)
+
+        cell.textLabel?.text = icon
+
+        if let imageView = cell.imageView {
+            let image = UIImage(named: previewImageName(for: icon))
+            imageView.image = image
+            imageView.layer.cornerRadius = Constants.cornerRadius
+            imageView.layer.masksToBounds = true
+            imageView.layer.borderColor = Constants.iconBorderColor?.cgColor
+            imageView.layer.borderWidth = borderedIcons.contains(icon) ? Constants.iconBorderWidth : 0
+        }
+
+        let isDefaultIconInUse = UIApplication.shared.alternateIconName == nil
+        if (isDefaultIconInUse && indexPath.row == 0) || UIApplication.shared.alternateIconName == icon {
+            cell.accessoryType = .checkmark
+        } else {
+            cell.accessoryType = .none
+        }
+
+        return cell
+    }
+
+    private func previewImageName(for icon: String) -> String {
+        let lowered = icon.lowercased().replacingMatches(of: " ", with: "_")
+        return "\(lowered)_\(AppIconViewController.iconBaseName)"
+    }
+
+    open override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let isOriginalIconRow = (indexPath.row == 0)
+        let icon = isOriginalIconRow ? nil : icons[indexPath.row]
+
+        UIApplication.shared.setAlternateIconName(icon, completionHandler: { [weak self] error in
+            if error == nil {
+                let event: WPAnalyticsStat = isOriginalIconRow ? .appIconReset : .appIconChanged
+                WPAppAnalytics.track(event)
+            }
+
+            self?.tableView.reloadData()
+        })
+    }
+
+    // MARK: - Private helpers
+
+    private func loadIcons() {
+        var icons = [Constants.defaultIconName]
+
+        // Load the names of the alternative app icons from the info plist
+        guard let iconDict = infoPlistIconsDict else {
+            self.icons = icons
+            return
+        }
+
+        // Add them (sorted) to the default key – first any prefixed with WordPress, then the rest.
+        let keys = iconDict.keys
+        let wordPressKeys = keys.filter({$0.hasPrefix("WordPress")}).sorted()
+        let otherKeys = keys.filter({!$0.hasPrefix("WordPress")}).sorted()
+        icons.append(contentsOf: (wordPressKeys + otherKeys))
```

Comment:
Not a big deal but if you treat `keys` as a _Set_, you can get the `otherKeys` subtracting `wordPressKeys`.
```
let keys = Set(iconDict.keys)
let wordPressKeys = keys.filter { $0.hasPrefix(_CONSTANT_PREFIX_) }.sorted()
let otherKeys = keys.subtracting(wordPressKeys).sorted()
```

Reply:
Ah, nice!